src/gui/dialogs/setupwizardpages/SetupThemePage.ts

```diff
@@ -1,27 +1,46 @@
 import m, { Children } from "mithril"
 import { WizardPageAttrs, WizardPageN } from "../../base/WizardDialog.js"
 import { lang } from "../../../misc/LanguageViewModel.js"
-import { RadioSelector, RadioSelectorAttrs } from "../../base/RadioSelector.js"
+import { RadioSelector, RadioSelectorAttrs, RadioSelectorOption } from "../../base/RadioSelector.js"
 import { themeController, themeOptions, ThemePreference } from "../../theme.js"
 import { SetupPageLayout } from "./SetupPageLayout.js"
 
-export class SetupThemePage implements WizardPageN<null> {
+export class SetupThemePage implements WizardPageN<SetupThemePageAttrs> {
+	// The whitelabel themes formatted as `RadioSelectorOption`s.
+	private customThemes: Array<RadioSelectorOption<string>> = []
```

Comment:
> you use two variables to represent the state: one for custom themes and one to indicate whether the themes are loaded. I am a big proponent of "make invalid states unrepresentable" so I would suggest pulling it into a single variable. It can be as simple as nullable array or it can be more explicit as `{state: "loading"} | {state: "loaded", themes: readonly string[] }` (I am not sure we need to prepare the options every time, to me it's part of view attributes that should be created on each render).

That is a better way of doing it for sure. I have gone the nullable array route now.